gradle-conjure-api/src/main/java/com/palantir/gradle/conjure/api/EndpointMinimumVersion.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * (c) Copyright 2021 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.gradle.conjure.api;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import java.io.Serializable;
+import java.util.Objects;
+
+public final class EndpointMinimumVersion implements Serializable {
```

Comment:
We tend to use immutable objects in most places rather than mutable bean style objects -- I think this project already use the immutables annotation processor, which may make it easier.

Reply:
I did that at first, but this object is (possibly) read in using gradle closure processing and I couldn't get that to work with Immutables because it uses direct reflection to set fields, not invoke the methods.  It technically works if you pass it a builder, but then the builder fails because the bitfield to indicate fields are set is not cleared.  I presume that's why the ServiceDependency object is the same.